src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesReconciliationService.kt

```diff
@@ -177,48 +177,53 @@ class CaseNotesReconciliationService(
 
   private val truncatedToSecondsFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss")
 
-  private fun CaseNote.transformFromDps() = ComparisonCaseNote(
-    id = caseNoteId,
-    text = text.truncateToFixNomisMaxLength(),
-    type = type,
-    subType = subType,
-    occurrenceDateTime = occurrenceDateTime.format(truncatedToSecondsFormatter),
-    creationDateTime = creationDateTime.format(truncatedToSecondsFormatter),
-    dpsUsername = authorUsername,
-    amendments = amendments.map { a ->
+  private fun CaseNote.transformFromDps(): ComparisonCaseNote {
+    val sortedAmendments = amendments.map { a ->
       CommonAmendmentFields(
-        text = a.additionalNoteText.truncateToFixNomisMaxLength(),
+        text = a.additionalNoteText,
         occurrenceDateTime = a.creationDateTime?.format(truncatedToSecondsFormatter),
         authorUsername = a.authorUserName,
       )
-    }.toSortedSet(amendmentComparator),
-    legacyId = legacyId,
-  )
+    }.toSortedSet(amendmentComparator)
+    return ComparisonCaseNote(
+      id = caseNoteId,
+      text = (text + sortedAmendments.joinToString("") { it.toNomisAmendment() }).truncateToFixNomisMaxLength(),
```

Comment:
Does this truncate mean that you don't include amendments in the comparison if the text is already max size?

Reply:
yes, exactly. That is what is happening at present though.  See https://prisoner-dev.digital.prison.service.justice.gov.uk/prisoner/G3101UO/case-notes for an example.  The amendments are missed off in the NOMIS database and only shown in DPS.